File: app/src/main/java/com/pluu/webtoon/adapter/MainFragmentAdapter.java
package com.pluu.webtoon.adapter;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.util.SparseArray;
import android.view.ViewGroup;

import com.pluu.support.impl.AbstractWeekApi;
import com.pluu.webtoon.common.Const;
import com.pluu.webtoon.ui.WebtoonListFragment;

/**
 * Main ViewPager Fragment Adapter
 * Created by PLUUSYSTEM-NEW on 2016-03-21.
 */
public class MainFragmentAdapter extends FragmentStatePagerAdapter {

    private final SparseArray<WebtoonListFragment> views = new SparseArray<>();
    private final AbstractWeekApi serviceApi;

    public MainFragmentAdapter(FragmentManager fm, AbstractWeekApi serviceApi) {
        super(fm);
        this.serviceApi = serviceApi;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        super.destroyItem(container, position, object);
        views.remove(position);
    }

    @Override
    public Fragment getItem(int position) {
        WebtoonListFragment fragment = new WebtoonListFragment();
        Bundle args = new Bundle();
        args.putInt(Const.EXTRA_POS, position);
        args.putSerializable(Const.EXTRA_API, serviceApi.getNaviItem());
        fragment.setArguments(args);

        views.put(position, fragment);

        return fragment;
    }

    @Override
    public int getCount() {
        return serviceApi.getWeeklyTabSize();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return serviceApi.getWeeklyTabName(position);
    }

}

File: app/src/main/java/com/pluu/webtoon/adapter/MainFragmentAdapter.kt
package com.pluu.webtoon.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.pluu.webtoon.domain.base.AbstractWeekApi
import com.pluu.webtoon.ui.weekly.WebtoonListFragment

/**
 * Main ViewPager Fragment Adapter
 * Created by pluu on 2017-05-02.
 */
class MainFragmentAdapter(
    fm: FragmentManager,
    lifecycle: Lifecycle,
    private val serviceApi: AbstractWeekApi
) : FragmentStateAdapter(fm, lifecycle) {

    override fun createFragment(position: Int): Fragment {
        return WebtoonListFragment.newInstance(position)
    }

    override fun getItemCount(): Int = serviceApi.weeklyTabSize
}
